DSLink/DSLinkContainer.cs

```diff
@@ -276,41 +224,57 @@ private void OnOpen()
             }
         }
 
-        /// <summary>
-        /// Event that is fired when a plain text message is sent to the broker.
-        /// </summary>
-        /// <param name="messageEvent">Text message event</param>
-        private void OnWrite(MessageEvent messageEvent)
+        private async void OnStringRead(MessageEvent messageEvent)
+        {
+            LogMessageString(false, messageEvent);
+            await OnMessage(DataSerializer.Deserialize(messageEvent.Message));
+        }
+
+        private void OnStringWrite(MessageEvent messageEvent)
+        {
+            LogMessageString(true, messageEvent);
+        }
+
+        private async void OnBinaryRead(BinaryMessageEvent messageEvent)
+        {
+            LogMessageBytes(false, messageEvent);
+            await OnMessage(DataSerializer.Deserialize(messageEvent.Message));
+        }
+
+        private void OnBinaryWrite(BinaryMessageEvent messageEvent)
+        {
+            LogMessageBytes(true, messageEvent);
+        }
+
+        private void LogMessageString(bool sent, MessageEvent messageEvent)
         {
             if (Logger.ToPrint.DoesPrint(LogLevel.Debug))
```

Comment:
I do this so that we don't waste any time building a string if we don't need to, even though it might be a very small amount of time.